python/grass/temporal/temporal_granularity.py

```diff
@@ -344,164 +423,83 @@ def compute_absolute_time_granularity(maps):
+            # Add to the set of the smallest granularity in the granularity_units dict
+            for time_unit in granularity_units:
+                if (
+                    time_unit in gap_datetime_delta
+                    and gap_datetime_delta[time_unit] > 0
+                ):
+                    granularity_units[time_unit].add(gap_datetime_delta[time_unit])
+                    break
+        # Keep the temporal extent to compare to the following/next map
+        previous_start, previous_end = start, end
 
     # Create a list with a single time unit only
-    if has_seconds:
-        for d in datetime_delta:
-            if "second" in d and d["second"] > 0:
-                dlist.append(d["second"])
-            elif "minute" in d and d["minute"] > 0:
-                dlist.append(d["minute"] * 60)
-            elif "hour" in d and d["hour"] > 0:
-                dlist.append(d["hour"] * 3600)
-            elif "day" in d and d["day"] > 0:
-                dlist.append(d["day"] * 24 * 3600)
-            else:
-                dlist.append(d["max_days"] * 24 * 3600)
-        use_seconds = True
-    elif has_minutes:
-        for d in datetime_delta:
-            if "minute" in d and d["minute"] > 0:
-                dlist.append(d["minute"])
-            elif "hour" in d and d["hour"] > 0:
-                dlist.append(d["hour"] * 60)
-            elif "day" in d:
-                dlist.append(d["day"] * 24 * 60)
-            else:
-                dlist.append(d["max_days"] * 24 * 60)
-        use_minutes = True
-    elif has_hours:
-        for d in datetime_delta:
-            if "hour" in d and d["hour"] > 0:
-                dlist.append(d["hour"])
-            elif "day" in d and d["day"] > 0:
-                dlist.append(d["day"] * 24)
-            else:
-                dlist.append(d["max_days"] * 24)
-        use_hours = True
-    elif has_days:
-        for d in datetime_delta:
-            if "day" in d and d["day"] > 0:
-                dlist.append(d["day"])
-            else:
-                dlist.append(d["max_days"])
-        use_days = True
-    elif has_months:
-        for d in datetime_delta:
-            if "month" in d and d["month"] > 0:
-                dlist.append(d["month"])
-            elif "year" in d and d["year"] > 0:
-                dlist.append(d["year"] * 12)
-        use_months = True
-    elif has_years:
-        for d in datetime_delta:
-            if "year" in d:
-                dlist.append(d["year"])
-        use_years = True
-
-    dlist.sort()
-    ulist = list(set(dlist))
-
-    if len(ulist) == 0:
+    dlist = set()
+    for time_unit, granularity_set in granularity_units.items():
+        # The smallest granularity unit is used so as soon as a non-zero
+        # value / set is encountered, the loop can be aborted
+        if granularity_set:
+            dlist.update(granularity_set)
+            break
+
```

Comment:
Before there was a handling of converting bigger "units" into the smaller one before gcd.
How does it create an equivalent effect now?
(And I'm still not convinced that `compute_datetime_delta` does what I think it should do, so don't let me bring you to a bad solution because of my comprehension).

Reply:
Yeah, seems I overshot the mark when simplified too much... Fixed in 04d4778.